src/braket/experimental/autoqasm/instructions/instructions.py

```diff
@@ -31,8 +42,43 @@ def _qubit_instruction(
     program_conversion_context.register_gate(name)
     program_conversion_context.register_args(args)
     program_mode = aq_program.ProgramMode.UNITARY if is_unitary else aq_program.ProgramMode.NONE
+    pos_control, neg_control = _get_pos_neg_control(control, control_state)
     oqpy_program = program_conversion_context.get_oqpy_program(mode=program_mode)
-    oqpy_program.gate([_qubit(q) for q in qubits], name, *args)
+    oqpy_program.gate(
+        [_qubit(q) for q in qubits],
+        name,
+        *args,
+        control=pos_control,
+        neg_control=neg_control,
+        exp=power,
+    )
+
+
+def _get_pos_neg_control(
+    control: QubitIdentifierType | Iterable[QubitIdentifierType] | None = None,
+    control_state: str | None = None,
+) -> tuple[Iterable[oqpy.Qubit], Iterable[oqpy.Qubit]]:
```

Comment:
Is there a benefit to a generic `Iterable` type in the return signature? It looks like we know the return type is always lists?

Reply:
Will change to `list`.